quipucords/tests/api/credential/test_credential.py

```diff
@@ -165,11 +211,74 @@ def test_hc_create_err_ssh_bad(self):
             "cred_type": DataSources.NETWORK,
             "ssh_keyfile": "blah",
         }
-        response = self.client.post(url, json.dumps(data), "application/json")
+        response = django_client.post(url, json=data)
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        assert response.json()["ssh_keyfile"]
+
+    def test_hc_create_err_ssh_keyfile_and_keyvalue(self, django_client):
+        """Test API with both ssh_keyfile and ssh_keyvalue.
+
+        Ensure we cannot create a new host credential object with both
+        an ssh_keyfile and ssh_keyvalue specified.
+        """
+        url = reverse("cred-list")
+        with tempfile.NamedTemporaryFile() as ssh_keyfile:
+            ssh_keyfile.write(b"\n")
+            data = {
+                "name": "cred1",
+                "username": "user1",
+                "cred_type": DataSources.NETWORK,
+                "ssh_keyfile": str(ssh_keyfile.name),
+                "ssh_keyvalue": "A" * 2048,
+            }
+            response = django_client.post(url, json=data)
```

Comment:
FYI there's a builtin pytest fixture for creation of temporary files: [tmp-path](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmp-path-fixture). It's a random directory and pytest will take care of cleaning that up after a test is done. I like this fixture because you can save the files for debugging passing a parameter to pytest (`--basetemp=somedir` ).

I don't see the need to use `tmp_path` fixture here, but I thought it would be cool to share the info.

T

Reply:
looks like `tmp_path` is already being used in this file, I'll update to use the same, we should be consistent.